ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -101,19 +138,43 @@ public StatsSnapshot getStatsSnapshot() throws StoreException {
    * 2. PUT expiring at t_exp but t_exp_ref < t_exp
    * 3. PUT with corresponding DELETE at time t_delete but t_del_ref < t_delete
    * 4. DELETE record
-   * For this API, t_del_ref is based on the given {@link TimeRange} and t_exp_ref is the time when the API is called.
+   * For this API, t_del_ref and t_exp_ref are the same and its value is based on the given {@link TimeRange}.
    * @param timeRange the reference {@link TimeRange} at which the data is requested. Defines both the reference time
    *                  and the acceptable resolution.
    * @return a {@link Pair} whose first element is the time at which stats was collected (in ms) and whose second
    * element is the valid data size for each segment in the form of a {@link NavigableMap} of segment names to
    * valid data sizes.
    */
   Pair<Long, NavigableMap<String, Long>> getValidDataSizeByLogSegment(TimeRange timeRange) throws StoreException {
-    long deleteReferenceTimeInMs = timeRange.getEndTimeInMs();
-    long expirationReferenceTimeInMs = time.milliseconds();
-    NavigableMap<String, Long> validSizePerLogSegment =
-        collectValidDataSizeByLogSegment(deleteReferenceTimeInMs, expirationReferenceTimeInMs);
-    return new Pair<>(deleteReferenceTimeInMs, validSizePerLogSegment);
+    ScanResults currentScanResults = scanResults.get();
+    Pair<Long, NavigableMap<String, Long>> retValue = null;
+    long deleteAndExpirationRefTimeInMs = getLogSegmentRefTimeMs(currentScanResults, timeRange);
+    if (deleteAndExpirationRefTimeInMs != -1) {
+      retValue = currentScanResults.getValidSizePerLogSegment(deleteAndExpirationRefTimeInMs);
```

Comment:
A instance/snapshot of ScanResults is taken at the beginning, all operation will be performed on that snapshot.